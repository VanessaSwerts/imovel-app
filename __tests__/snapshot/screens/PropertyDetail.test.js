import React from 'react'
import { create } from 'react-test-renderer'
import PropertyDetail from '../../../src/screens/PropertyDetail'

import mockedProperty from '../../mocks/constants/property.json'

jest.runAllTimers()
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: jest.fn()
    }),
    useRoute: () => ({
      route: () => ({
        params: mockedProperty
      })
    })
  }
})

describe('PropertyDetail snapshot test', () => {
  it('render PropertyDetail screen correctly', () => {
    const tree = create(<PropertyDetail />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
