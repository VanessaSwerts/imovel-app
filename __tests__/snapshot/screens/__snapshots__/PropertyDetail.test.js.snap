// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyDetail snapshot test render PropertyDetail screen correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#0565FC",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bounces={false}
    scrollsToTop={true}
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#2C3E50",
            "fontSize": 20,
            "fontWeight": "700",
            "height": 280,
            "justifyContent": "center",
            "paddingHorizontal": 30,
            "paddingTop": 110,
            "textAlign": "center",
          }
        }
      >
        Nenhum imagem disponibilizada para este imóvel!
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderTopLeftRadius": 50,
            "flex": 1,
            "height": "100%",
            "marginTop": -15,
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Object {
              "color": "#2C3E50",
              "fontSize": 22,
              "fontWeight": "700",
              "paddingBottom": 5,
              "paddingTop": 10,
              "textAlign": "right",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#AAADB3",
              "fontSize": 17,
              "fontWeight": "500",
              "lineHeight": 20,
              "textAlign": "right",
            }
          }
        >
          , 
        </Text>
        <Text
          style={
            Object {
              "color": "#AAADB3",
              "fontSize": 17,
              "fontWeight": "500",
              "lineHeight": 20,
              "textAlign": "right",
            }
          }
        >
           - 
        </Text>
        <Text
          style={
            Object {
              "color": "#2C3E50",
              "fontSize": 18,
              "fontWeight": "600",
              "paddingBottom": 5,
              "paddingTop": 20,
            }
          }
        >
          Sobre esse imóvel:
        </Text>
        <Text
          numberOfLines={5}
          style={
            Object {
              "color": "#AAADB3",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 18,
              "marginBottom": 10,
              "textAlign": "justify",
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 20,
              "marginTop": 20,
              "width": "80%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#7F8C8D",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "elevation": 1,
                  "height": 45,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "shadowColor": "#2C3E50",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.05,
                  "shadowRadius": 1,
                  "width": 45,
                }
              }
            >
              <Icon
                color="#7F8C8D"
                name="bed"
                size={18}
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#7F8C8D",
                }
              }
            >
               quartos
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#7F8C8D",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "elevation": 1,
                  "height": 45,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "shadowColor": "#2C3E50",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.05,
                  "shadowRadius": 1,
                  "width": 45,
                }
              }
            >
              <Icon
                color="#7F8C8D"
                name="bath"
                size={20}
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#7F8C8D",
                }
              }
            >
               banheiro
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 20,
              "marginTop": 20,
              "width": "80%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#7F8C8D",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "elevation": 1,
                  "height": 45,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "shadowColor": "#2C3E50",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.05,
                  "shadowRadius": 1,
                  "width": 45,
                }
              }
            >
              <Icon
                color="#7F8C8D"
                name="ruler-horizontal"
                size={18}
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#7F8C8D",
                }
              }
            >
               m²
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#7F8C8D",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "elevation": 1,
                  "height": 45,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "shadowColor": "#2C3E50",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.05,
                  "shadowRadius": 1,
                  "width": 45,
                }
              }
            >
              <Icon
                color="#7F8C8D"
                name="dog"
                size={19}
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#7F8C8D",
                }
              }
            >
              Não aceita pet
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 20,
              "marginTop": 15,
              "width": "95%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#2C3E50",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "paddingBottom": 5,
                  "paddingTop": 20,
                }
              }
            >
              Aluguel:
            </Text>
            <Text
              numberOfLines={5}
              style={
                Object {
                  "color": "#AAADB3",
                  "fontSize": 15,
                  "fontWeight": "500",
                  "paddingLeft": 8,
                }
              }
            >
              
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-end",
                "backgroundColor": "#0565FC",
                "borderRadius": 8,
                "height": 35,
                "justifyContent": "center",
                "opacity": 1,
                "width": 160,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Entrar em contato
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "left": 15,
            "opacity": 1,
            "position": "absolute",
            "top": 15,
            "zIndex": 5,
          }
        }
      >
        <Icon
          color="#2C3E50"
          name="arrow-left"
          size={20}
        />
      </View>
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#e6e6e6",
                "borderColor": "#0565FC",
                "borderRadius": 24,
                "borderWidth": 0.5,
                "elevation": 5,
                "padding": 18,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginBottom": 5,
                }
              }
            >
              <Image
                resizeMode="contain"
                source={1}
                style={
                  Object {
                    "backgroundColor": "#0565FC",
                    "borderColor": "#0565FC",
                    "borderRadius": 100,
                    "borderWidth": 3,
                    "height": 80,
                    "width": 80,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "justifyContent": "center",
                    "paddingLeft": 8,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#0565FC",
                        "fontSize": 14,
                        "fontWeight": "700",
                        "paddingBottom": 5,
                        "paddingRight": 5,
                      }
                    }
                  >
                    Proprietário:
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#6e6e6e",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "paddingBottom": 5,
                      }
                    }
                  >
                    Bianca Souza
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#0565FC",
                        "fontSize": 14,
                        "fontWeight": "700",
                        "paddingBottom": 5,
                        "paddingRight": 5,
                      }
                    }
                  >
                    Telefone:
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#6e6e6e",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "paddingBottom": 5,
                      }
                    }
                  >
                    (35) 98877-6655
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#0565FC",
                        "fontSize": 14,
                        "fontWeight": "700",
                        "paddingBottom": 5,
                        "paddingRight": 5,
                      }
                    }
                  >
                    E-mail:
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#6e6e6e",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "paddingBottom": 5,
                      }
                    }
                  >
                    biancasouza@gmail.com
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-end",
                  "backgroundColor": "#2C3E50",
                  "borderRadius": 8,
                  "height": 30,
                  "justifyContent": "center",
                  "width": 80,
                }
              }
              testID="modal-button"
            >
              <Text
                style={
                  Object {
                    "color": "#0565FC",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  }
                }
              >
                Fechar
              </Text>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
